inlong-sort/sort-core/src/main/java/org/apache/inlong/sort/parser/impl/NativeFlinkSqlParser.java

```diff
@@ -0,0 +1,93 @@
+ *   with the License.  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.apache.inlong.sort.parser.impl;
+
+import com.google.common.base.Preconditions;
+import org.apache.flink.table.api.TableEnvironment;
+import org.apache.inlong.sort.function.RegexpReplaceFirstFunction;
+import org.apache.inlong.sort.parser.Parser;
+import org.apache.inlong.sort.parser.result.FlinkSqlParseResult;
+import org.apache.inlong.sort.parser.result.ParseResult;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * parse flink sql script file
+ * script file include CREATE TABLE statement
+ */
+public class NativeFlinkSqlParser implements Parser {
+
+    private static final Logger log = LoggerFactory.getLogger(FlinkSqlParser.class);
+
+    private final TableEnvironment tableEnv;
+    private final String statements;
+
+    public NativeFlinkSqlParser(TableEnvironment tableEnv, String statements) {
+        this.tableEnv = tableEnv;
+        this.statements = statements;
+        registerUDF();
+    }
+
+    /**
+     * Register udf
+     */
+    private void registerUDF() {
+        tableEnv.createTemporarySystemFunction("REGEXP_REPLACE_FIRST", RegexpReplaceFirstFunction.class);
+    }
+
+    /**
+     * Get a instance of NativeFlinkSqlParser
+     *
+     * @param tableEnv The tableEnv,it is the execution environment of flink sql
+     * @param statements The statements, it is statement set
+     * @return NativeFlinkSqlParser The flink sql parse handler
+     */
+    public static NativeFlinkSqlParser getInstance(TableEnvironment tableEnv, String statements) {
+        return new NativeFlinkSqlParser(tableEnv, statements);
+    }
+
+
+    /**
+     * parse flink sql script file
+     *
+     * @return ParseResult the result of parsing
+     */
+    @Override
+    public ParseResult parse() {
+        Preconditions.checkNotNull(statements, "sql statement set is null");
+        Preconditions.checkNotNull(tableEnv, "tableEnv is null");
+        List<String> createTableSqls = new ArrayList<>();
+        List<String> insertSqls = new ArrayList<>();
+        String[] statementSet = statements.split(";");
+        for (String statement : statementSet) {
+            statement = statement.trim();
+            if (statement.startsWith("CREATE TABLE") || statement.startsWith("create table")) {
+                createTableSqls.add(statement);
+            } else if (statement.startsWith("INSERT INTO") || statement.startsWith("insert into")) {
```

Comment:
It is recommended to ignore case when comparing strings.

Reply:
> It is recommended to ignore case when comparing strings.

done